src/BenchmarkDotNet/Engines/GcStats.cs

```diff
@@ -136,17 +137,16 @@ public static GcStats ReadFinal()
         public static GcStats FromForced(int forcedFullGarbageCollections)
             => new GcStats(forcedFullGarbageCollections, forcedFullGarbageCollections, forcedFullGarbageCollections, 0, 0);
 
+        // Make sure tier0 jit doesn't cause any unexpected allocations in this method.
+        [MethodImpl(CodeGenHelper.AggressiveOptimizationOption)]
         private static long? GetAllocatedBytes()
         {
             // we have no tests for WASM and don't want to risk introducing a new bug (https://github.com/dotnet/BenchmarkDotNet/issues/2226)
             if (RuntimeInformation.IsWasm)
                 return null;
 
-            // "This instance Int64 property returns the number of bytes that have been allocated by a specific
-            // AppDomain. The number is accurate as of the last garbage collection." - CLR via C#
-            // so we enforce GC.Collect here just to make sure we get accurate results
-            GC.Collect();
-
+            // Calling GC.Collect() before calling GC.GetTotalAllocatedBytes appears to interfere with the results for some reason,
```

Comment:
I checked the docs, and garbage collection was not mentioned for `MonitoringTotalAllocatedMemorySize`. It's only mentioned for `MonitoringSurvivedMemorySize`. `GC.GetTotalAllocatedBytes` docs also don't mention it. But sure, we can wait for @Maoni0's input.

Either way, I'm not sure what caused the interference. Is there possibly something listening to GC events causing allocations somewhere?

Reply:
from GC's POV, there's no relation between reporting allocated bytes and doing a `GC.Collect`, in other words, GC doesn't adjust the allocated bytes in any way in the implementation of `GC.Collect`.

however, you might see a difference if you call GC.Collect vs not because GC.Collect may cause some app behavior to change, for example, maybe some threads will become dead and so [`Thread::dead_threads_non_alloc_bytes`](https://github.com/dotnet/runtime/blob/f27056529982d9aea44f0a84f24fc0bb0ee52c96/src/coreclr/vm/comutilnative.cpp#L985) is adjusted. whether this actually happened in your case or not, I don't know. that's just a guess.